systems/plants/RigidBodyManipulator.h

```diff
@@ -284,6 +284,8 @@ class DLLEXPORT_RBM RigidBodyManipulator
   std::string getBodyOrFrameName(int body_or_frame_id);
   //@param body_or_frame_id   the index of the body or the id of the frame.
 
+  int parseBodyOrFrameID(const int body_or_frame_id, Matrix4d* Tframe = nullptr);
```

Comment:
I actually recommended it to reduce code duplication. Maybe we should add an out of bounds check to make things a little safer. But passing in frame indices to kinematics methods is already part of the public interface, so having the logic for converting those to body indices and frame transforms be a public method seems fine (even necessary) to me.